draft-ZIP-0226.rst

```diff
@@ -0,0 +1,303 @@
+
+Note that this is enough to create a chain of induction that ensures that all output notes of a transfer are actual outputs of a GroupHash, preventing any malleability attacks, as they originate in the Issuance protocol, which is publicly verified. Furthermore, we do not care about whether the note is owned by the sender, or whether it was nullified before. Wallets and other clients have a choice to make to ensure the asset type is the preserved for the output note of a split Action, for the value balance verification:
+
+1. The split input note could be the same note as the original (non-split) Action, 
+2. The split input note could be a different unspent note of the same type (note that the note will not actually be spent)
+3. The split input note could be an already spent note of the same type (note that by zeroing the value in the circuit, we prevent double spending)
+
+The specific circuit changes are presented below.
+
+Circuit Statement
+=================
+
+The advantage of the design described above, with respect to the circuit statement, is that every *ZSA Action statement* is kept closely similar to the Orchard Action statement [#protocol-actionstatement]_, except for a few additions that ensure the security of the asset type system.
+
+**Asset Type Equality:** the following constraints must be added to ensure that the input and output note are of the
+same type:
+
+- The asset type, :math:`\mathsf{type_\mathbb{P}}`, for the note is witnessed once, as an auxiliary input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the old note commitment input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the new note commitment input.
+
+**Correct Value Commitment Type:** the following constraints must be added to ensure that the value commitment is computed using the witnessed type, as represented in the notes
+
+- The fixed-base multiplication constraints between the value and the value base point of the value commitment,:math:`\mathsf{cv}`, is replaced with a variable-base multiplication between the two
+- The witness to the value base-point is the auxiliary input :math:`\mathsf{type}_\mathbb{P}`.
+
+**Enforce Secure Type for Split Actions:** the following constraints must be added to prevent senders from malling with the asset type for the output note in the consequent actions of the split:
+
+- The Value Commitment Integrity should be changed
+    - Replace the input note value by a generic value, `v'`, as :math:`\mathsf{cv^net} = \mathsf{ValueCommit_rcv^OrchardType(v’ - v^new, type}_\mathbb{P})`
+- Add a boolean “split” variable as an auxiliary witness. This variable is to be activated `split = 1` if the Action in question is a split and `split = 0` if the Action is actually spending an input note:
+    - If `split = 1` then set `v' = 0` otherwise `v'=v^old` from the auxiliary input
```

Comment:
```suggestion
    - If :math:`split = 1` then constrain :math:`v' = 0` otherwise constrain :math:`v'=v^old` from the auxiliary input.
    - If :math:`split = 1` then constrain :math:`v^{old} \neq 0`.
```
The latter condition is necessary in order to ensure that the Merkle Path Validity condition is still enforced, so that we know that the input note has a valid base.

Reply:
@PaulLaux @ConstanceBeguier we need to ensure this is in the implementation.

Reply:
Change incorporated into the ZIP.